### PR TITLE
Move database initialization to corresponding bean

### PR DESCRIPTION
**Pull request by diega**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 18 issues
* ![MAJOR][MAJOR] 14 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L115](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/co/rsk/Start.java#L115): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Start.java#L121](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/co/rsk/Start.java#L121): Method co.rsk.Start.enableRpc(Rsk) needlessly boxes a boolean constant [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOOLEAN_CONSTANT_CONVERSION)
1. ![MAJOR][MAJOR] [RskFactory.java#L74](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L74): Method has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskFactory.java#L124](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L124): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L49](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L49): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L51](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L51): Method org.ethereum.datasource.mapdb.MapDBDataSource.init() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L51](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L51): Exceptional return value of java.io.File.mkdirs() ignored in org.ethereum.datasource.mapdb.MapDBDataSource.init() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ARV_RETURN_VALUE_IGNORED_BAD_PRACTICE)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L90](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L90): Method org.ethereum.datasource.mapdb.MapDBDataSource.put(byte[], byte[]) uses array as basis of collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AABC_ARRAY_BASED_COLLECTIONS)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L114](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L114): Iterate over the "entrySet" instead of the "keySet". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2864)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L115](https://github.com/rsksmart/rskj/blob/638997e04b68d89622443bd60a8a94b839f97d16/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L115): org.ethereum.datasource.mapdb.MapDBDataSource.updateBatch(Map) makes inefficient use of keySet iterator instead of entrySet iterator [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AWMI_WRONG_MAP_ITERATOR)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 18 issues
* ![MAJOR][MAJOR] 14 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L117](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/co/rsk/Start.java#L117): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Start.java#L123](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/co/rsk/Start.java#L123): Method co.rsk.Start.enableRpc(Rsk) needlessly boxes a boolean constant [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOOLEAN_CONSTANT_CONVERSION)
1. ![MAJOR][MAJOR] [RskFactory.java#L74](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L74): Method has 14 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskFactory.java#L125](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L125): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L49](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L49): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L51](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L51): Method org.ethereum.datasource.mapdb.MapDBDataSource.init() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L51](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L51): Exceptional return value of java.io.File.mkdirs() ignored in org.ethereum.datasource.mapdb.MapDBDataSource.init() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ARV_RETURN_VALUE_IGNORED_BAD_PRACTICE)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L90](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L90): Method org.ethereum.datasource.mapdb.MapDBDataSource.put(byte[], byte[]) uses array as basis of collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AABC_ARRAY_BASED_COLLECTIONS)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L114](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L114): Iterate over the "entrySet" instead of the "keySet". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2864)
1. ![MAJOR][MAJOR] [MapDBDataSource.java#L115](https://github.com/rsksmart/rskj/blob/40d5804d4e181871487ccab5ec23826974439705/rskj-core/src/main/java/org/ethereum/datasource/mapdb/MapDBDataSource.java#L115): org.ethereum.datasource.mapdb.MapDBDataSource.updateBatch(Map) makes inefficient use of keySet iterator instead of entrySet iterator [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AWMI_WRONG_MAP_ITERATOR)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 47 issues
* ![MAJOR][MAJOR] 34 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 10 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L68](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/co/rsk/Start.java#L68): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L70](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L70): Method has 14 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [CommonConfig.java#L74](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/config/CommonConfig.java#L74): Method org.ethereum.config.CommonConfig.wireTransactions() passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L96](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L96): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L98](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L98): Method org.ethereum.config.DefaultConfig.blockStore() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L98](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L98): Exceptional return value of java.io.File.mkdirs() ignored in org.ethereum.config.DefaultConfig.blockStore() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ARV_RETURN_VALUE_IGNORED_BAD_PRACTICE)
1. ![MAJOR][MAJOR] [DataSourcePool.java#L29](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/datasource/DataSourcePool.java#L29): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L68](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L68): Method new org.ethereum.datasource.LevelDbDataSource(String) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L75](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L75): Method org.ethereum.datasource.LevelDbDataSource.init() passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L98](https://github.com/rsksmart/rskj/blob/e64bb8ebbbc4521de9ff1c443db6fdc3770816be/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L98): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 42 issues
* ![MAJOR][MAJOR] 29 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 10 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L68](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/co/rsk/Start.java#L68): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L70](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L70): Method has 14 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [DefaultConfig.java#L95](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java#L95): java/io/File.<init>(Ljava/lang/String;)V reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [DataSourcePool.java#L29](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/datasource/DataSourcePool.java#L29): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L98](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L98): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L99](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L99): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L99](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L99): Method org.ethereum.datasource.LevelDbDataSource.init() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L101](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L101): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L112](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L112): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [LevelDbDataSource.java#L112](https://github.com/rsksmart/rskj/blob/7d30dc63ed73c12ec37c141a7d3977828a276972/rskj-core/src/main/java/org/ethereum/datasource/LevelDbDataSource.java#L112): Method org.ethereum.datasource.LevelDbDataSource.init() throws alternative exception from catch block without history [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALEST_LOST_EXCEPTION_STACK_TRACE)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java [hunk 1]
```diff
@@ -86,10 +85,13 @@ public void init() {
...
-        String blocksIndexFile = database + "/blocks/index";
-        File dbFile = new File(blocksIndexFile);
-        if (!dbFile.getParentFile().exists()) {
-            dbFile.getParentFile().mkdirs();
+        File blockIndexDirectory = new File(database + "/blocks/");
+        File dbFile = new File(blockIndexDirectory, "index");
+        if (!blockIndexDirectory.exists()) {
+            boolean mkdirsSuccess = blockIndexDirectory.mkdirs();
+            if (!mkdirsSuccess) {
+                logger.error("Unable to create blocks directory: {}", blockIndexDirectory);
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java]: Shouldn't we throw? I can't see how we can recover from this error

**Pull request merged**